src/skmatter/_selection.py

```diff
@@ -1038,7 +1038,14 @@ def _init_greedy_search(self, X, y, n_to_select):
         self.hausdorff_ = np.full(X.shape[self._axis], np.inf)
         self.hausdorff_at_select_ = np.full(X.shape[self._axis], np.inf)
 
-        if self.initialize == "random":
+        if isinstance(self.initialize, np.ndarray):
```

Comment:
Shouldn't these operattions also be performed for a List and not only for a numpy array?

Reply:
Yes. Let me know what you think about the new changes I made.